content/articles/integrate-ml-to-flask-api-and-deploy-to-heroku/index.md

```diff
@@ -0,0 +1,336 @@
+            new_record = [[pregnancies, glucose, bp, st, insulin, bmi, dpf, age]]
+            # feature scale the data
+            scaled_data = sc.transform(new_record)
+            # dictionary containing the diagnosis with the key as the model's prediction
+            diagnosis = {0: 'Your Result is Normal', 
+                         1: 'Diabetes Detected'
+                        }
+            # pass scaled data to model for prediction
+            new_pred = model.predict(scaled_data)[0]
+            # get corresponding value from the diagnosis dictionary (using the model prediction as the key)
+            result = diagnosis.get(new_pred)
+            return {'Diagnosis': result}, 200
+        except:
+            # if client sends the wrong request or data type then return correct format
+            return {'Error! Please use this JSON format': {"Pregnancies": 5, 
+                                              "Glucose": 32,
+                                              "BloodPressure": 43,
+                                              "SkinThickness": 23,
+                                              "Insulin":22,
+                                              "BMI": 33.6,
+                                              "DiabetesPedigreeFunction":  1.332,
+                                              "Age": 50 
+                                             }}, 500
+```
+
+Finally, we add the URI of the `Records` resource and run the API.
+
+```python
+api.add_resource(Records, '/')
+app.run(port=5000, debug=True)
+```
+
+We test the local API with Postman to make sure it works before deploying it. 
+
+![GET request to local server](/engineering-education/integrate-ml-to-flask-api-and-deploy-to-heroku/get_request_local.PNG)
+
+After sending a GET request, we get a response showing the JSON format we should send in a POST request. So now we send a POST request with JSON data in the same format shown above. 
+
+![POST request to local server](/engineering-education/integrate-ml-to-flask-api-and-deploy-to-heroku/post_request_to_local_api.PNG)
+
+After sending the POST request, we get the model prediction as a response. Let us see what happens if the client sends the wrong JSON data to the API.
+
+![POST request with the wrong format to local server](/engineering-education/integrate-ml-to-flask-api-and-deploy-to-heroku/post_wrong_request_to_local_server.PNG)
+
+If we send the wrong JSON data to the API, we get an error message with the 500 status code. However, the client will also receive the correct JSON data format. Since our API is working fine, we can proceed to deploy it.
+
+
+### Deploying to Heroku
+
+The next step is to deploy our API so other developers can have access to it. Next, we will deploy the API to Heroku. Heroku is a cloud platform that allows developers to build, run, and deploy their apps. If you do not have a Heroku account, you will have to create one by clicking [here](https://signup.heroku.com/).

```

Comment:
If you do not have a Heroku account, you can create one by clicking here.

Reply:
Done!